src/Bicep.Core.UnitTests/Parser/ParserTests.cs

```diff
@@ -216,6 +216,13 @@ public void PropertyAccessShouldParseSuccessfully(string text, string expected)
             RunExpressionTest(text, expected, typeof(PropertyAccessSyntax));
         }
 
+        [DataTestMethod]
+        [DataRow("a.b.c.foo()", "((a.b).c)foo()")]
```

Comment:
>foo [](start = 42, length = 3)

The lack of . in the expected expression actually made me realize that round-tripping isn't working right. You should update one of the Invalid* samples to use the instance function syntax.